src/pre_process/include/2dHardcodedIC.fpp

```diff
@@ -1,10 +1,49 @@
 #:def Hardcoded2DVariables()
-
+!> @file  2DHardcodedIC.fpp
+!> @brief Initialize a 2D grid by projecting a 1D profile.
+!>
+!> @details
+!> Reads a sequence of 1D primitive-variable files and maps them onto
+!> a 2D domain along the x-direction. These files are produced when parallel I/O is disabled in `case.py`.
+!> Parameters control file counts, grid resolution, and domain offsets.
+!>
+!> @param nFiles        Number of primitive-variable files to read
+!> @param nRows         Number of grid points per row
+!> @param init_dir      Directory containing the `prim.*.dat` files
     real(wp) :: eps
     real(wp) :: r, rmax, gam, umax, p0
     real(wp) :: rhoH, rhoL, pRef, pInt, h, lam, wl, amp, intH, intL, alph
     real(wp) :: factor
 
+    integer, parameter :: nFiles = 14   ! Number of files (variables)
+    integer, parameter :: nRows = 401   ! Number of grid points
+    integer :: f, iter, ios, unit, idx, jump, index_1
+    real(wp) :: x_len, x_step
+    integer :: global_offset  ! MPI subdomain offset
+    real(wp) :: delta_x
+    character(len=100), dimension(nFiles) :: fileNames
+    character(len=200) :: errmsg
+    ! Arrays to store all data from files
+    real(wp), dimension(nRows, nFiles) :: stored_values  ! Imported Data
+    real(wp), dimension(nRows) :: x_coords
+    logical :: files_loaded = .false.
+    real(wp) :: domain_start, domain_end
+    character(len=*), parameter :: init_dir = "/home/pain/MFC-Adam/examples/1D_reactive_shocktube/D/"
```

Comment:
you can't leave it like this i don't think... also since it's a parameter the user can't change it without recompiling, right?